testcases/cloud_admin/modify_instance_type.py

```diff
@@ -0,0 +1,176 @@
+                  user_data=None, instance_user=None, **kwargs):
+
+        super(InstanceTypeOps, self).__init__(name=name)
+        self.get_args()
+        #self.show_args()
+        for kw in kwargs:
+            print 'Setting kwarg:'+str(kw)+" to "+str(kwargs[kw])
+            self.set_arg(kw ,kwargs[kw])
+        #self.show_args()
+        self.setuptestcase()
+        self.setup_parser()
+        #self.get_args()
+        # Setup basic eutester object
+        self.tester = Eucaops(config_file=self.args.config_file,
+                                  password=self.args.password)
+                                  # credpath=self.args.credpath)  # <- credpath disabled since not functioning for these commands
+
+        self.PopulateInstanceTypes()
+        #print "Now printing _my_types after populating:"    # DEBUG
+        #for i in InstanceTypeOps._known_types:
+            #i.p()
+
+    def clean_method(self):
+        pass
+
+    def check_output_type(self, arg=None):
+        """
+        :param arg: argument must be a list or it will assert
+        :return: assert if not list
+        """
+        assert type(arg) != None, "Output from self.sys() empty!"
+        try:
+            not isinstance(arg, basestring)
+        except TypeError:
+            AssertionError("Output from self.sys() should be list!")
+        #print "Output type checks out correctly!"    # DEBUG
+
+    def DescribeInstanceTypes(self, item=None):
+
+        if item is None:
+            cmd = "source " + self.tester.credpath + "/eucarc && /usr/bin/euca-describe-instance-types"
+        else:
+            cmd = "source " + self.tester.credpath + "/eucarc && /usr/bin/euca-describe-instance-types " + item
+
+        print "in DescribeInstanceTypes function Running " + cmd
+
+        output = self.tester.sys(cmd)
+
+        self.check_output_type(output)
+
+        return output
+
+    def ModifyInstanceType(self, item=None):
+
+        if item is None:
+            raise ValueError("Incorrect number of arguments, must pass cmd to ModifyInstanceType()")
+        else:
+            cmd = "source " + self.tester.credpath + "/eucarc && /usr/bin/euca-modify-instance-type " + ' '.join(item)
+
+
+        print "in ModifyInstanceType function Running " + cmd
+        output = self.tester.sys(cmd)
+        self.check_output_type(output)
+        return output
+
+    def PopulateInstanceTypes(self):
+        print "Populating cluster Instance Types..."
+        output = self.DescribeInstanceTypes()
+
+        for row in output:
+            #print "Got output: " + row    # DEBUG
+            if "INSTANCETYPE" in row:
+                items = re.split("\s+", row)[1:5]
+                if items[1] != "CPUs":   # ignore the heading line
+                    InstanceTypeOps._known_types.append(instance_type(name=items[0],cpus=int(items[1]),
+                                                                  mem=int(items[2]),disk=int(items[3])))
+
+    def Modify(self):
```

Comment:
Lots of these methods modify/populate/describe will be generally useful in other testcases. I would like to see these be put into the Eucaops class and called from the testcase.